### PR TITLE
JmsServiceBuilder

### PR DESCRIPTION
**Pull request by BenjaminBarrett**

### CODE REVIEW

File: gateway/src/com/kamino/go/jms/event/RequestMessageBody.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	@Override
+	public Map<String, Object> getBody() {
+		Map<String, Object> body = new HashMap<>();
+		body.put("body", getRequestBody());
+		body.put("pathParameters", getPathParameters());
+		return body;
+	}
+	
+	private Map<String, Object> getRequestBody() {
+		if(getRequestContext() instanceof ContainerRequest) {
```
Comment (Degran) [inline gateway/src/com/kamino/go/jms/event/RequestMessageBody.java]: Seems like an opportunity for polymorphism. Should look into what the possible results of getRequestContext() are and what the non-ContainerRequests are for.

File: gateway/src/com/kamino/go/jms/event/RequestMessageBody.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+		return convertToMap(getRequestContext().getUriInfo()
+							  				   .getPathParameters());
+	}
+	
+	private Map<String, Object> convertToMap(MultivaluedMap<String, String> multivaluedMap) {
+		return multivaluedMap.entrySet().stream()
+							 .collect(Collectors.toMap(Entry::getKey, Entry::getValue));
+	}
+
+}
```
Comment (Degran) [inline gateway/src/com/kamino/go/jms/event/RequestMessageBody.java]: Why so much indirection with RequestEvent, RequestMessageBody, GetRequestBody, GetBody, body, ...
Could at least use some more creative and diverse naming :p

File: gateway/src/com/kamino/go/modules/JmsServiceModule.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package com.kamino.go.modules;
+
+import com.google.inject.AbstractModule;
+import com.google.inject.Singleton;
+import com.kamino.go.jms.JmsService;
+import com.kamino.go.jms.JmsServiceProvider;
+
+public class JmsServiceModule extends AbstractModule {
```
Comment (Degran) [inline gateway/src/com/kamino/go/modules/JmsServiceModule.java]: Small pet-peeve: I would usually avoid naming such as AbstractModule.
Nobody should care that it is abstract. They'll find out when they try to instantiate it.
On the other hand, it looks a bit weird when you're handling objects polymorphically and their types are 'Abstract'. For this reason, I would sooner name something 'Implementation' than 'Abstract' or 'Interface'.

Comment (BenjaminBarrett) [reply gateway/src/com/kamino/go/modules/JmsServiceModule.java]: import com.google.inject.AbstractModule
It's a guice class

Comment (BenjaminBarrett) [reply gateway/src/com/kamino/go/modules/JmsServiceModule.java]: But generally I'd agree

File: gateway/src/com/kamino/go/jms/event/RequestMessageBody.java (hunk 2 above)
Comment (BenjaminBarrett) [reply gateway/src/com/kamino/go/jms/event/RequestMessageBody.java]: I'll revisit this code. My main goal was to split jms vs REST functionality, it's a sad coincidence "POST" data is called the request body.

Comment (BenjaminBarrett) [reply gateway/src/com/kamino/go/jms/event/RequestMessageBody.java]: I like the distinction between Event and Message (Renamed MessageBody). Right now event contains no information Message does not provide but it allows flexibility in the future.

**Pull request merged**